tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -2444,8 +2455,14 @@ void SwiftLangSupport::editorClose(StringRef Name, bool RemoveCache) {
     IFaceGenContexts.remove(Name);
   }
 
-  if (Removed && RemoveCache)
-    Removed->removeCachedAST();
+  if (Removed) {
+    // Cancel any in-flight builds for the given AST.
+    Removed->cancelBuildsForCachedAST();
+
+    // Then remove the cached AST if we've been asked to do so.
+    if (RemoveCache)
+      Removed->removeCachedAST();
```

Comment:
Is there any point to not removing from the cache if we never re-use it anyway? Or is there some case where it is re-used?